### PR TITLE
Removed Inline Styles For DevSite Examples

### LINKED ISSUE #498: Remove inline styles in dev site components
# Feature Request

## Description

We want to remove inline styles from our dev site components to set a better example for consumers of terra.  As a replacement, we should pass in classes and use css to style the components.

### PR DESCRIPTION
**Pull request by lokesh-0813**
### Summary
Replaced Inline Style in JSX components for DEVSITE with SCSS files

resolves #498

### CODE REVIEW

**Comment by StephenEsser** [comment]
There are a few other inline examples within other components. 

We released a new version of eslint-config-terra yesterday that warns when it detects an inline style. If you `npm run clean:install` you should get the updated package. Running `npm run lint` will log these warnings in the terminal. This should make finding them a little easier.

File: packages/terra-clinical-data-grid/src/terra-dev-site/doc/clinical-data-grid/DatagridExamples.2/DatagridWithCustomContent.jsx [hunk 1]
```diff
@@ -115,7 +115,7 @@ class DatagridWithCustomContent extends React.Component {
 
   render() {
     return (
-      <div style={{ height: '800px' }}>
+      <div className={cx('data-grid-div-style')}>
```
Comment (ShettyAkarsh) [inline packages/terra-clinical-data-grid/src/terra-dev-site/doc/clinical-data-grid/DatagridExamples.2/DatagridWithCustomContent.jsx]: we can keep the name as 'data-grid-parent-div-style'

File: packages/terra-clinical-data-grid/src/terra-dev-site/doc/clinical-data-grid/DatagridExamples.2/DatagridWithSubsections.scss [hunk 2]
```diff
@@ -0,0 +1,5 @@
+:local {
+  .data-grid-div-style {
```
Comment (ShettyAkarsh) [inline packages/terra-clinical-data-grid/src/terra-dev-site/doc/clinical-data-grid/DatagridExamples.2/DatagridWithSubsections.scss]: We can create a common css for this one. as the height value is common across most of the examples.

File: packages/terra-clinical-header/src/terra-dev-site/doc/example/HeaderLongTextWithContent.scss [hunk 3]
```diff
@@ -0,0 +1,5 @@
+:local {
```
Comment (ShettyAkarsh) [inline packages/terra-clinical-header/src/terra-dev-site/doc/example/HeaderLongTextWithContent.scss]: even this one can be reused.

File: packages/terra-clinical-data-grid/src/terra-dev-site/doc/clinical-data-grid/DatagridExamples.2/Datagrid.jsx [hunk 4]
```diff
@@ -74,7 +77,7 @@ class StaticDataGrid extends React.Component {
 
   render() {
     return (
-      <div style={{ height: '800px' }}>
+      <div className={cx('data-grid-div-style')}>
```
Comment (StephenEsser) [inline packages/terra-clinical-data-grid/src/terra-dev-site/doc/clinical-data-grid/DatagridExamples.2/Datagrid.jsx]: We should follow a similar approach to naming the example classes as we would for released components. I think we should avoid using the associate tag (`div`) and redundant (`style`) within the name. This class for example could be `data-grid-example` or something that describes what this example component is.

File: packages/terra-clinical-data-grid/src/terra-dev-site/doc/clinical-data-grid/DatagridExamples.2/DatagridWithCustomContent.jsx (hunk 1 above)
Comment (lokesh-0813) [reply packages/terra-clinical-data-grid/src/terra-dev-site/doc/clinical-data-grid/DatagridExamples.2/DatagridWithCustomContent.jsx]: Classnames are renamed.

File: packages/terra-clinical-data-grid/src/terra-dev-site/doc/clinical-data-grid/DatagridExamples.2/Datagrid.jsx (hunk 4 above)
Comment (lokesh-0813) [reply packages/terra-clinical-data-grid/src/terra-dev-site/doc/clinical-data-grid/DatagridExamples.2/Datagrid.jsx]: Classnames have been renamed with component names for better understanding.

**Comment by lokesh-0813** [comment]
> I think that this PR would be a lot cleaner if, especially for `DataGrid`, each test just referenced `_Datagrid.scss` and there was a single class with `height: 800px;` instead of each test having it's own `.scss` file with it's own single class in it.

@Blackop778 It was intentionally written in this approach to support any future modifications. For instance in `terra-grid` , there are scenarios where height is the only property but in few scenarios there is more than just height. Any upcoming modifications can be approached in ease.

**Comment by StephenEsser** [comment]
I think the shared / extended SCSS for examples is becoming complicated. I'd like to get people's thoughts on removing any shared scss. Each example has its own scss file with no extended styles. 

Here is the primary thoughts I have on this:

We may eventually display the example scss to the user on the doc site. That may end up looking something like this:

image

When the users clicks to display the scss with the current approach on this PR they would end up seeing something similar to this.

image

This makes the actual styles of the example abstracted away from the user and no longer provides much value.

But if the user was able to see the SCSS at face value they would be able to see the styles applied to the example:

image

**Comment by emilyrohrbough** [comment]
I agree with @StephenEsser. Although `extends` is probably good practice in src code, I think using `extends` in our doc examples will make it harder for consumers to easily determine what styles where applied on a example.

**Comment by StephenEsser** [comment]
Do you have any thoughts on this @bjankord ?

**Comment by bjankord** [comment]
+1 to "removing any shared scss. Each example has its own scss file with no extended styles."
-1 to "We may eventually display the example scss to the user on the doc site." I don't believe that we'd want to display the styles in the doc site as we have been considering these implementation details. If people start to copy and paste styles out of terra components, those implementation details become more and more leaky.

**Comment by StephenEsser** [comment]
The styles are shown to the user within the examples today with the use of inline styles.  My thoughts are that if the examples require css to display correctly, those implementation details should also be available to the user. Ideally the example code and the css to create that example are paired together. Copy pasting has always been pretty a tricky issue, I'll add this topic to the war room notes to talk through the pros and cons of hiding / showing the example styles a little more depth.

**Comment by ShettyAkarsh** [comment]
@StephenEsser , @bjankord, @emilyrohrbough  should we go ahead and make this changes for the example files only ? by removing the shared scss  and keep the terra-dev-site->test folder shared scss implementation as its is ? for which we are not displaying the scss implementation for the user.

File: packages/terra-clinical-data-grid/src/terra-dev-site/doc/clinical-data-grid/DatagridExamples.2/Datagrid.scss [hunk 5]
```diff
@@ -0,0 +1,8 @@
+@import './Datagrid';
+
+:local {
+  .data-grid-basic {
+    @extend %data-grid-height;
```
Comment (emilyrohrbough) [inline packages/terra-clinical-data-grid/src/terra-dev-site/doc/clinical-data-grid/DatagridExamples.2/Datagrid.scss]: Lets remove the @extend syntax and add the styles direclty.

File: packages/terra-clinical-data-grid/src/terra-dev-site/doc/clinical-data-grid/DatagridExamples.2/Datagrid.scss [hunk 6]
```diff
@@ -0,0 +1,8 @@
+@import './Datagrid';
```
Comment (emilyrohrbough) [inline packages/terra-clinical-data-grid/src/terra-dev-site/doc/clinical-data-grid/DatagridExamples.2/Datagrid.scss]: Lets drop the DataGrid import and add the styles directory or import both sass files into the example and class bind as needed

File: packages/terra-clinical-data-grid/src/terra-dev-site/doc/clinical-data-grid/DatagridExamples.2/DatagridWithColumnResizing.scss [hunk 7]
```diff
@@ -0,0 +1,7 @@
+@import './Datagrid';
```
Comment (emilyrohrbough) [inline packages/terra-clinical-data-grid/src/terra-dev-site/doc/clinical-data-grid/DatagridExamples.2/DatagridWithColumnResizing.scss]: This is one-to-one styles to `Datagrid.scss`. Lets reuse `Datagrid.scss` and remove this file.

File: packages/terra-clinical-data-grid/src/terra-dev-site/doc/clinical-data-grid/DatagridExamples.2/DatagridWithNoPinnedColumns.scss [hunk 8]
```diff
@@ -0,0 +1,7 @@
+@import './Datagrid';
```
Comment (emilyrohrbough) [inline packages/terra-clinical-data-grid/src/terra-dev-site/doc/clinical-data-grid/DatagridExamples.2/DatagridWithNoPinnedColumns.scss]: This is one-to-one styles to `Datagrid.scss`. Lets reuse `Datagrid.scss` and remove this file.

File: packages/terra-clinical-data-grid/src/terra-dev-site/doc/clinical-data-grid/DatagridExamples.2/DatagridWithSelections.scss [hunk 9]
```diff
@@ -0,0 +1,7 @@
+@import './Datagrid';
```
Comment (emilyrohrbough) [inline packages/terra-clinical-data-grid/src/terra-dev-site/doc/clinical-data-grid/DatagridExamples.2/DatagridWithSelections.scss]: zThis is one-to-one styles to `Datagrid.scss`. Lets reuse `Datagrid.scss` and remove this file.

File: packages/terra-clinical-data-grid/src/terra-dev-site/doc/clinical-data-grid/DatagridExamples.2/DatagridWithSubsections.scss [hunk 10]
```diff
@@ -0,0 +1,7 @@
+@import './Datagrid';
+
```
Comment (emilyrohrbough) [inline packages/terra-clinical-data-grid/src/terra-dev-site/doc/clinical-data-grid/DatagridExamples.2/DatagridWithSubsections.scss]: This is one-to-one styles to `Datagrid.scss`. Lets reuse `Datagrid.scss` and remove this file.

File: packages/terra-clinical-data-grid/src/terra-dev-site/doc/clinical-data-grid/DatagridExamples.2/_Datagrid.scss [hunk 11]
```diff
@@ -0,0 +1,5 @@
+:local {
```
Comment (emilyrohrbough) [inline packages/terra-clinical-data-grid/src/terra-dev-site/doc/clinical-data-grid/DatagridExamples.2/_Datagrid.scss]: lets remove this file and add these styles directly

File: packages/terra-clinical-data-grid/src/terra-dev-site/doc/clinical-data-grid/DatagridExamples.2/DatagridWithCustomContent.scss [hunk 12]
```diff
@@ -0,0 +1,12 @@
+@import './Datagrid';
```
Comment (emilyrohrbough) [inline packages/terra-clinical-data-grid/src/terra-dev-site/doc/clinical-data-grid/DatagridExamples.2/DatagridWithCustomContent.scss]: IMO these docs styles can be merged into a common example with class names that are meaningful to their intent

File: packages/terra-clinical-detail-view/src/terra-dev-site/doc/example/_Detailview.scss [hunk 13]
```diff
@@ -0,0 +1,5 @@
+:local {
```
Comment (emilyrohrbough) [inline packages/terra-clinical-detail-view/src/terra-dev-site/doc/example/_Detailview.scss]: Lets create a common sass file to share across the examples and lets drop the @extends syntax

File: packages/terra-clinical-header/src/terra-dev-site/doc/example/HeaderLongTextWithContent.scss (hunk 3 above)
Comment (emilyrohrbough) [reply packages/terra-clinical-header/src/terra-dev-site/doc/example/HeaderLongTextWithContent.scss]: Lets create a common sass file to share across the examples and lets drop the @extends syntax

**Comment by lokesh-0813** [comment]
@emilyrohrbough  @ShettyAkarsh @StephenEsser @Blackop778 
Updated examples to use styles from a common style file.
Removed `@extends` and `imports` implementation.

Also JEST test is failing for terra-onset-picker, hence updated OnsetPicker.test.jsx.snap.

File: packages/terra-clinical-detail-view/src/terra-dev-site/doc/example/DetailViewDividedSmallerTitles.jsx [hunk 14]
```diff
@@ -23,7 +26,7 @@ const detail2item4 = (<LabelValueView label="Admin Notes" textValue="Try med for
 
 
 const DetailViewDivided = () => (
-  <div style={{ border: '3px dotted #d3d4d5' }}>
+  <div className={cx('detail-view-divided-small-title')}>
```
Comment (emilyrohrbough) [inline packages/terra-clinical-detail-view/src/terra-dev-site/doc/example/DetailViewDividedSmallerTitles.jsx]: I think this class name needs updated to `detail-view-divided`

File: packages/terra-clinical-detail-view/src/terra-dev-site/doc/example/DetailViewNoDivider.jsx [hunk 15]
```diff
@@ -2,14 +2,18 @@ import React from 'react';
...
+const cx = classNames.bind(styles);
 
 const item1 = (<LabelValueView label="Start Date" textValue="12/12/12 00:00AM" />);
 const item2 = (<LabelValueView label="Stop Date" textValue="12/12/12 10:00AM" />);
 const item3 = (<LabelValueView label="Collected By" textValue="Nuse Collect" />);
 const item4 = (<LabelValueView label="Priority" textValue="High" />);
 
 const DetailViewDivided = () => (
-  <div style={{ border: '3px dotted #d3d4d5' }}>
+  <div className={cx('detail-view-no-divider')}>
```
[the remark this answers is not in the document]
Comment (lokesh-0813) [reply packages/terra-clinical-detail-view/src/terra-dev-site/doc/example/DetailViewNoDivider.jsx]: Yes @emilyrohrbough, I missed on that. I have them updated now. Thank you.

File: packages/terra-clinical-data-grid/src/terra-dev-site/doc/clinical-data-grid/DatagridExamples.2/Datagrid.scss [hunk 16]
```diff
@@ -0,0 +1,20 @@
+:local {
```
Comment (emilyrohrbough) [inline packages/terra-clinical-data-grid/src/terra-dev-site/doc/clinical-data-grid/DatagridExamples.2/Datagrid.scss]: @yuderekyu @bjankord shouldn't these sass files be named `name.module.scss`?

Comment (bjankord) [reply packages/terra-clinical-data-grid/src/terra-dev-site/doc/clinical-data-grid/DatagridExamples.2/Datagrid.scss]: Left a comment on a similar PR in terra-core for this: https://github.com/cerner/terra-core/pull/2465#discussion_r295826939

Comment (lokesh-0813) [reply packages/terra-clinical-data-grid/src/terra-dev-site/doc/clinical-data-grid/DatagridExamples.2/Datagrid.scss]: Renamed name.scss to name.module.scss.

File: packages/terra-clinical-onset-picker/CHANGELOG.md [hunk 17]
```diff
@@ -3,6 +3,9 @@ ChangeLog
 
 Unreleased
 ----------
+### Changed
```
Comment (dkasper-was-taken) [inline packages/terra-clinical-onset-picker/CHANGELOG.md]: Is this change log entry needed, I don't see any corresponding changes. Otherwise looks good.

**Pull request merged**